server/app/views/addTask.scala.html

```diff
@@ -3,17 +3,18 @@
 
 @(task: Form[AddTaskForm])(implicit msg: Messages)
 
-@main(msg("addTask")) {
-    <div><a href="@routes.Application.index()">@msg("home")</a></div>
-    <div class="solutionForm">
-    @form(routes.NewTask.postNewTask()) {
-        @textarea(task(taskDescription), 'rows -> "8", 'cols -> "80")
-        @textarea(task(solutionTemplate), 'rows -> "6", 'cols -> "80")
-        @textarea(task(referenceSolution), 'rows -> "13", 'cols -> "80")
-        @textarea(task(suite), 'rows -> "13", 'cols -> "80")
-        <div class="submit">
-            <input type="submit" value="@msg("addTask")"/>
+    @main(msg("addTask")) {
+        <div><a href="@routes.Application.index()">@msg("home")</a></div>
+        <div class="solutionForm">
+        @form(routes.NewTask.postNewTask()) {
+            @textarea(task(taskDescription), 'rows -> "8", 'cols -> "80")
+            @textarea(task(solutionHeader), 'rows -> "3", 'cols -> "80")
```

Comment:
//header
class SubArrayWithMaxSum extends SubArrayWithMaxSumSolution {
  def apply(a: Array[Int]): Array[Int] = {
//body
    var currentSum = 0
    var maxSum = 0
    var left, right = 0
    var maxI = 0 //used when all negatives in the array

    for (i <- a.indices) {
      val incSum = currentSum + a(i)

      if (incSum > 0) {
        currentSum = incSum

        if (currentSum > maxSum) {
          maxSum = currentSum
          right = i
        }
      } else {
        left = i + 1
        right = left
        currentSum = 0
        if (a(i) > a(maxI)) maxI = i
      }
    }

    if (left == a.length) a.slice(maxI, maxI + 1)
    else a.slice(left, right + 1)
//footer
  }
}